cpu/cc2538/dev/spi.c

```diff
@@ -41,102 +41,337 @@
+#error "Both SPI0_RX_PORT and SPI0_RX_PIN must be valid or invalid"
+#endif
+
+/* Here we check that either all or none of the ports are defined. As
+   we did already check that both ports + pins are either defined or
+   not for every pin, this means that we can check for an incomplete
+   configuration by only looking at the port defines */
+#if (SPI0_CLK_PORT < 0 && SPI0_TX_PORT < 0 && SPI0_RX_PORT >= 0) || \
+  (SPI0_CLK_PORT < 0 && SPI0_TX_PORT >= 0 && SPI0_RX_PORT < 0) || \
+  (SPI0_CLK_PORT < 0 && SPI0_TX_PORT >= 0 && SPI0_RX_PORT >= 0) || \
+  (SPI0_CLK_PORT >= 0 && SPI0_TX_PORT < 0 && SPI0_RX_PORT < 0) || \
+  (SPI0_CLK_PORT >= 0 && SPI0_TX_PORT < 0 && SPI0_RX_PORT >= 0) || \
+  (SPI0_CLK_PORT >= 0 && SPI0_TX_PORT >= 0 && SPI0_RX_PORT < 0)
+#error "SPI0 port / pin definition incomplete"
+#endif
+/*---------------------------------------------------------------------------*/
+/* Check port / pin settings for SPI1 and provide default values for spi_cfg */
+#ifndef SPI1_CLK_PORT
+#define SPI1_CLK_PORT         (-1)
+#endif
+#ifndef SPI1_CLK_PIN
+#define SPI1_CLK_PIN          (-1)
+#endif
+#if SPI1_CLK_PORT >= 0 && SPI1_CLK_PIN < 0 || \
+  SPI1_CLK_PORT < 0 && SPI1_CLK_PIN >= 0
+#error "Both SPI1_CLK_PORT and SPI1_CLK_PIN must be valid or invalid"
+#endif
+
+#ifndef SPI1_TX_PORT
+#define SPI1_TX_PORT          (-1)
+#endif
+#ifndef SPI1_TX_PIN
+#define SPI1_TX_PIN             (-1)
+#endif
+#if SPI1_TX_PORT >= 0 && SPI1_TX_PIN < 0 || \
+  SPI1_TX_PORT < 0 && SPI1_TX_PIN >= 0
+#error "Both SPI1_TX_PORT and SPI1_TX_PIN must be valid or invalid"
+#endif
+
+#ifndef SPI1_RX_PORT
+#define SPI1_RX_PORT            (-1)
+#endif
+#ifndef SPI1_RX_PIN
+#define SPI1_RX_PIN             (-1)
+#endif
+#if SPI1_RX_PORT >= 0 && SPI1_RX_PIN < 0 || \
+  SPI1_RX_PORT < 0 && SPI1_RX_PIN >= 0
+#error "Both SPI1_RX_PORT and SPI1_RX_PIN must be valid or invalid"
+#endif
+
+/* Here we check that either all or none of the ports are defined. As
+   we did already check that both ports + pins are either defined or
+   not for every pin, this means that we can check for an incomplete
+   configuration by only looking at the port defines */
+#if (SPI1_CLK_PORT < 0 && SPI1_TX_PORT < 0 && SPI1_RX_PORT >= 0) || \
+  (SPI1_CLK_PORT < 0 && SPI1_TX_PORT >= 0 && SPI1_RX_PORT < 0) || \
+  (SPI1_CLK_PORT < 0 && SPI1_TX_PORT >= 0 && SPI1_RX_PORT >= 0) || \
+  (SPI1_CLK_PORT >= 0 && SPI1_TX_PORT < 0 && SPI1_RX_PORT < 0) || \
+  (SPI1_CLK_PORT >= 0 && SPI1_TX_PORT < 0 && SPI1_RX_PORT >= 0) || \
+  (SPI1_CLK_PORT >= 0 && SPI1_TX_PORT >= 0 && SPI1_RX_PORT < 0)
+#error "SPI1 port / pin definition incomplete"
+#endif
+/*---------------------------------------------------------------------------*/
+typedef struct {
+  int8_t port;
+  int8_t pin;
+} spi_pad_t;
+typedef struct {
+  uint32_t base;
+  uint32_t ioc_ssirxd_ssi;
+  uint32_t ioc_pxx_sel_ssi_clkout;
+  uint32_t ioc_pxx_sel_ssi_txd;
+  spi_pad_t clk;
+  spi_pad_t tx;
+  spi_pad_t rx;
+} spi_cfg_t;
```

Comment:
Please name this `spi_regs_t` for consistency with the UART driver. This is mostly the definition of the registers to use for one SPI instance, even if for the pads this comes from a configuration.